swap out rethink-knex-db migrations in tests with knexfile.js migration

schuyler1d opened the pull request:
Besides merging main, this adds https://github.com/harpojaeger/Spoke/commit/9facda8a6f40e143a2fc009f7158070f74699544
which replaces the test harness with your migration.

The tests seem to run nicely!

Commit:
documentation first

schuyler1d committed:
implement cached opt-outs per-organization

schuyler1d committed:
wip: campaignContacdt cache loading and structure. TODO: messageStatus handling with load, clear of course

schuyler1d committed:
Merge remote-tracking branch 'origin/main' into 813-decouple-contactgetting

schuyler1d committed:
await assignmentRequired

schuyler1d committed:
Merge remote-tracking branch 'origin/813-decouple-contactgetting' into scaling-campaigncontacts2

schuyler1d committed:
load,clear for campaign-contact. TODO: messageStatus

lperson committed:
Merge branch 'set_twilio_message_validity' into staging20180902

joemcl committed:
Merge pull request #54 from lperson/staging20180902

merge lperson/set_twilio_message_validity into staging20180902

lperson committed:
remove duplicate imports

joemcl committed:
Merge pull request #55 from lperson/staging20180902

update with lperson/Spoke - remove duplicate imports

lperson committed:
skip creating the message if the resolver receives it after texting hours

lperson committed:
Merge branch 'set_twilio_message_validity' into staging20180902

lperson committed:
Merge remote-tracking branch 'moveon/main' into set_twilio_message_validity

lperson committed:
fixing merge conflict

lperson committed:
Merge branch 'main' into set_twilio_message_validity

lperson committed:
fix merge error

shakalee14 committed:
Merge branch '863-datawarehouse-imports-cell-formatting' into stage-main-2

joemcl committed:
Merge pull request #56 from bchrobot/lock-send-messages

merging bchrobot/Spoke - Lock send messages

schuyler1d committed:
fix contact load

schuyler1d committed:
Merge remote-tracking branch 'origin/813-decouple-contactgetting' into scaling-campaign

schuyler1d committed:
simplify getContacts() conditions

schuyler1d committed:
correct assignment sql tests to match sql tweaks

shakalee14 committed:
Merge pull request #845 from lperson/remove_unnecessary_npm_target_and_associated_file

Clean up cruft related to server debugging support

mathemagica committed:
admin UI in settings for updating org-specific opt-out message defaults, new api query and resolver

mathemagica committed:
instead of generic error message on error parsing exception, still display the raw message as a string

mathemagica committed:
Merge branch 'main' into 741-default-instancewide-optout-message

shakalee14 committed:
Reverts changes in 722

schuyler1d committed:
Merge remote-tracking branch 'origin/main' into scaling-campaign

schuyler1d committed:
campaign caching spike

mathemagica committed:
resolve opt out message override with env-wide default, move this to api/organization and out of view

schuyler1d committed:
export campaign in cacheableData

shakalee14 committed:
Merge pull request #864 from MoveOnOrg/stage-main-2

Stage main 2

schuyler1d committed:
merge main into scaling-campaign

schuyler1d committed:
fix merge conflict

schuyler1d committed:
add cacheableData README.md stub

schuyler1d committed:
organization cache clearing

schuyler1d committed:
documenting caching environment variables

shakalee14 committed:
WIP: passing orgId based on env var setting

schuyler1d committed:
load() call needs to return a real Campaign() object rather than a dejsoned object

mathemagica committed:
more explanation of how batching and windowing work for contact loading optimization

schuyler1d committed:
Merge remote-tracking branch 'origin/main' into 817-sqlitefix

schuyler1d committed:
dueBy rendered as date regardless of object value

schuyler1d committed:
Merge remote-tracking branch 'origin/817-sqlitefix' into scaling-tip

shakalee14 committed:
Merge branch '817-sqlitefix' into 884-instance-opt-outs

shakalee14 committed:
Merge branch 'scaling-tip' into 884-instance-opt-outs

mathemagica committed:
tweaks

mathemagica committed:
tweaks

Commit:
some distinction between zero and null

Commit:
saving zeroes but not blanks

shakalee14 committed:
Removing opts outs based on env setting for uploads and warehouse querying

Commit:
now saving blank max contacts

Commit:
remove some logs and disable Send First Texts if max contacts 0

joemcl committed:
Merge pull request #59 from MoveOnOrg/779-killing-jobs

merging MoveOn/Spoke - 779 killing jobs

shakalee14 committed:
WIP: making sure opting out contact in one campaign disables texting them in other campaign in other org

Commit:
cleanup

shakalee14 committed:
Adds documentation for env var

schuyler1d committed:
merge lperson/new_texting_hours_per_campaign into main

lperson committed:
Merge remote-tracking branch 'moveon/main' into message_review_for_ipad

azuzunaga committed:
Add hasUnsentInitialMessages to the schema

azuzunaga committed:
Query the DB for unsent messages

schuyler1d committed:
merge origin/scaling-tip with new_texting_hours_per_campaign

azuzunaga committed:
Add tag letting admin know if a campaign has unsent initial messages

shakalee14 committed:
Reverting changes in campaign contact to reduce complexity

shakalee14 committed:
Allowing logic to be based on env var vs orgId

shakalee14 committed:
Removes duplicate code and creates global func for opt out queries

mathemagica committed:
Merge pull request #892 from MoveOnOrg/813-decouple-contactgetting-moredocs

more explanation of how batching and windowing work for contact loading optimization

mathemagica committed:
Merge branch 'main' into 741-default-instancewide-optout-message

mathemagica committed:
pr feedback

Commit:
added sorting to post-scaling query to load canned responses from db

shakalee14 committed:
Changes func name to match what it does - returns cells not count

shakalee14 committed:
Merge branch 'cp4r3z-617-e2e-edit-user' into stage-main

shakalee14 committed:
Merge branch 'force-display-of-graphql-errors' into stage-main

shakalee14 committed:
Merge branch '817-sqlitefix' into stage-main

shakalee14 committed:
Merge branch '741-default-instancewide-optout-message' into stage-main

shakalee14 committed:
Merge branch 'migrate-ifnotdbcreate' into stage-main

shakalee14 committed:
Merge branch 'scaling-tip' into stage-main

shakalee14 committed:
Merge branch '813-decouple-contactgetting' into stage-main

shakalee14 committed:
Merge branch 'simplify-getcontacts' into stage-main

shakalee14 committed:
Merge branch '828-fix-maxContacts' into stage-main

shakalee14 committed:
Merge branch '884-instance-opt-outs' into stage-main

shakalee14 committed:
Merge branch '798-response-sort-scaling' into stage-main

shakalee14 committed:
Merge branch 'new_texting_hours_per_campaign-withcaching' into stage-main

azuzunaga committed:
Add a new title color to warn admins when a campaign has unsent initial messages

azuzunaga committed:
Only display unsent messages tag when the campaign has started

schuyler1d committed:
dur, need to return cached campaign object

shakalee14 committed:
Merge branch 'scaling-tip' into stage-main

schuyler1d committed:
bump rethink-knex-adapter -- allow saving an object with extra properties

shakalee14 committed:
Merge branch 'scaling-tip-savefix' into stage-main

shakalee14 committed:
Change default to 50 for initial page load

shakalee14 committed:
Only check for listSize and pageSize if campaignFilters exist

shakalee14 committed:
Merge branch 'campaign-page-related-fixes' into stage-main

lperson committed:
fixed tests after bad merge

lperson committed:
fixed tests after bad merge

mathemagica committed:
also invalidate org cache after updating opt out message

schuyler1d committed:
tweak csv to have multiple timezones in the file

shakalee14 committed:
Changes default listSize to 100

shakalee14 committed:
Adds hash to fix link and remove 404 error

shakalee14 committed:
Merge pull request #913 from MoveOnOrg/fixing-lambda-deploy-docs-link

Adds hash to fix link and remove 404 error

schuyler1d committed:
fix top script, which was not the initial one, but sometimes the last one!

shakalee14 committed:
Merge pull request #912 from MoveOnOrg/testdata-in2timezones

tweak csv to have multiple timezones in the file

schuyler1d committed:
fix contactsFilter undefined case in getContacts

schuyler1d committed:
make sure we actually have an existing contact in AssignmentTexter componentWillUpdate

schuyler1d committed:
fix loading a new contact info after a dynamic assignment

schuyler1d committed:
lint AssignmentTexter.jsx

lperson committed:
ui for reassign all matching, make reassign buttons disabled if no reassign texter is selected

lperson committed:
ui for reassign all matching, make reassign buttons disabled if no reassign texter is selected

lperson committed:
reassign all matching conversations

lperson committed:
reassign all matching conversations

lperson committed:
tweak

lperson committed:
tweak

joemcl committed:
Merge pull request #62 from lperson/wfp_reassign_all_matching

reassign all matching conversations

Commit:
stub assignment test

shakalee14 committed:
Merge pull request #901 from MoveOnOrg/stage-main

Stage main

lperson committed:
filter for opt-out in conversations query

lperson committed:
indicate in message review which contacts are opted out

lperson committed:
tweak

azuzunaga committed:
Merge branch 'main' into 875-visual-texts-assigned-not-sent

lperson committed:
filter by opt-out status fully implemented

lperson committed:
Merge branch 'message_review_opt_out' into wfp_message_review_opt_out

joemcl committed:
Merge pull request #64 from lperson/wfp_message_review_opt_out

merging lperson/Spoke message_review_opt_out

lperson committed:
Merge branch 'set_twilio_message_validity' into merge_moveon_set_twilio_message_validity

lperson committed:
force a new build

lperson committed:
Revert "force a new build"

This reverts commit b12a9e1

Commit:
more tests, these should work on updated master or scaling-tip

Commit:
Merge branch 'main' into assignment-test

Commit:
super long timeout to see if that's the prob on travis

shakalee14 committed:
Merge pull request #933 from MoveOnOrg/assignment-test

Some tests of assignTexters

shakalee14 committed:
Return empty string instead of null to temp surpass malformatted cell nos

shakalee14 committed:
Removing invalid cell nos from count

shakalee14 committed:
Merge branch 'main' into 909-db-querying-v2

azuzunaga committed:
Merge branch 'main' of https://github.com/MoveOnOrg/Spoke into 875-visual-texts-assigned-not-sent

jparkrr committed:
Admin ability to filter people by campaign

jparkrr committed:
Avoid error if no campaign is found

jparkrr committed:
Ability to return roles from campaign.texters

jparkrr committed:
Revert "Avoid error if no campaign is found"

This reverts commit 86ede798f19cb78f8cbbe3311497523124160605.

jparkrr committed:
Revert "Ability to return roles from campaign.texters"

This reverts commit 2f845b32c92d3f420671c9876698a882b1fccd01.

jparkrr committed:
Ability to filter org users by campaignId

jparkrr committed:
Use organization query to get campaign-filtered users

shakalee14 committed:
Adds message for why were returning empty str vs null

shakalee14 committed:
Does correct check if count returns 0 in order to update result_message in jobrequest table

Commit:
error logging

lperson committed:
Merge remote-tracking branch 'moveon/main' into reassign_all_matching

lperson committed:
merge

lperson committed:
Merge branch 'moveon_main_plus_reassign_all_matching' into moveon_main_plus_reassign_all_plus_message_review_optout

lperson committed:
fixed a couple of merge issues

lperson committed:
Merge branch 'moveon_main_plus_reassign_all_plus_message_review_optout' into moveon_main_plus_reassign_all_plus_message_review_optout_plus_set_message_validity

lperson committed:
merge fixes

lperson committed:
Merge branch 'moveon_main_plus_reassign_all_plus_message_review_optout_plus_set_message_validity' into wfp_staging20180902

joemcl committed:
Merge pull request #72 from lperson/moveon_main_plus_reassign_all_plus_message_review_optout_plus_set_message_validity

Moveon main plus reassign all plus message review optout plus set message validity

joemcl committed:
Revert "Moveon main plus reassign all plus message review optout plus set message validity"

joemcl committed:
Merge pull request #73 from WorkingFamilies/revert-72-moveon_main_plus_reassign_all_plus_message_review_optout_plus_set_message_validity

Revert "Moveon main plus reassign all plus message review optout plus set message validity"

lperson committed:
fix interaction step

lperson committed:
Merge branch 'staging20180902' into moveon_main_plus_reassign_all_plus_message_review_optout_plus_set_message_validity

joemcl committed:
Merge pull request #74 from lperson/moveon_main_plus_reassign_all_plus_message_review_optout_plus_set_message_validity

fix interaction step

shakalee14 committed:
Add opted out check on hasUnsentInitialMsgs

shakalee14 committed:
Attempting to fix travis build

shakalee14 committed:
Fixes merge conflict for updated customFields

jparkrr committed:
Run tests in one process

Some tests are dependant on this. Later on we can seperate them out.

jparkrr committed:
Add forceexit to jest

Some async process isn't being cleaned up correctly. Probably the real
solution is to eventually look into that

shakalee14 committed:
Merge pull request #964 from jparkrr/jp-fix-tests

WIP: Fix tests

jparkrr committed:
Run e2e tests last

jparkrr committed:
Fix e2e tests

jparkrr committed:
Add bail option to jest for e2e

shakalee14 committed:
Merge pull request #953 from MoveOnOrg/stage-main-2

Stage main 2

shakalee14 committed:
WIP - figuring out how to pass validation stats

shakalee14 committed:
Merge branch 'main' into 909-db-querying-v2 and fix server errs

shakalee14 committed:
Fix let const constraints

Commit:
new job to add users to default org

shakalee14 committed:
Runs lint on main

Commit:
doc'ed this temp fix

shakalee14 committed:
Merge pull request #968 from MoveOnOrg/run-lint-on-main

Runs lint on main

jparkrr committed:
Merge branch 'main' into jp-admin-filter-users

shakalee14 committed:
Only gets timezone_offset when present and batch insert sz to pass to rka

shakalee14 committed:
Merge branch 'main' into 909-db-querying-v2

shakalee14 committed:
Changes option prop to batchSize for clarity purposes

shakalee14 committed:
Bumps rethink-knex-adapter version for batchSize option

shakalee14 committed:
Merge branch 'jparkrr-jp-admin-filter-users' into stage-main

shakalee14 committed:
Merge branch '934-orgless-user' into stage-main

shakalee14 committed:
adds beta files to gitignore

jparkrr committed:
Fix role change by passing the campaign id

shakalee14 committed:
Merge pull request #977 from MoveOnOrg/fix-role-change

Fix role change in admin filter people by campaign

shakalee14 committed:
Merge pull request #976 from MoveOnOrg/stage-main

Stage main

schuyler1d committed:
opportunistically open the warehouse connection -- otherwise, dont connect until/if we need it

schuyler1d committed:
Merge pull request #988 from MoveOnOrg/warehouseconnect-opportunisticmain

opportunistically open the warehouse connection -- otherwise, dont co…

schuyler1d committed:
accept twilio messages with special characters: a workaround from Twilio sending X-Twilio-Body with the full message and AWS/Lambda/express library rejecting the header values as invalid

schuyler1d committed:
delete duplicates after the job is complete -- this guards against multiple jobs and other accidents

schuyler1d committed:
fix lints in delete area and add "AS c2"

schuyler1d committed:
add more data to loadContactsFromDataWarehouseFragment logs

yeti-detective committed:
fix for issue 818

schuyler1d committed:
add a test for lambda.js -- the AWS platform

schuyler1d committed:
add a test for lambda.js -- the AWS platform

schuyler1d committed:
Merge pull request #993 from MoveOnOrg/lambda-acceptspecialchartexts

accept twilio messages with special characters

lperson committed:
apply paginated campaigns fix to revision running in production

joemcl committed:
Merge pull request #76 from lperson/apply_paginated_campaigns_fix_to_revision_running_in_production

apply paginated campaigns fix to revision running in production

shakalee14 committed:
change page size to all

schuyler1d committed:
creating an org and then joining -- we only want to save a single role now

schuyler1d committed:
creating an org and then joining -- we only want to save a single role now

lperson committed:
fix the problem where clicking view messages unselects selected conversations

joemcl committed:
Merge pull request #77 from lperson/fix_view_messages_unselects

fix the problem where clicking view messages unselects selected conversations

lperson committed:
changed refreshing messages when some contacts go out of texting hours

joemcl committed:
Merge pull request #78 from lperson/fix_view_messages_unselects

changed message when some contacts go out of texting hours

lperson committed:
Merge branch 'moveon_main_12_2' into wfp_staging_20180902

lperson committed:
Revert "Revert "Moveon main plus reassign all plus message review optout plus set message validity""

This reverts commit 5da1dcd552a0b9c4cd4fa7b28733d4145320e27e.

lperson committed:
fixed a broken test

joemcl committed:
Adding link to ORM Q and A b/c not developer (#1027)

joemcl committed:
Adding cross-reference to Exports doc (#1026)

* Adding cross-reference to Exports doc

Added link to https://github.com/MoveOnOrg/Spoke/blob/main/docs/DATA_EXPORTING.md .

* Update EMAIL_CONFIGURATION.md

* removed typos

joemcl committed:
Sample SQL for a dashboard: counts of texters, sent, responses, total texts, opt outs (#1021)

* counts of texters, sent, responses, total texts, opt outs

added 
* counts of texters, sent, responses, total texts, opt outs and estimated cost for all current campaigns

* updated sample query, added explanation

updated sample query, added explanation of segments and link to https://www.twilio.com/blog/2017/03/what-the-heck-is-a-segment.html#segment

shakalee14 committed:
Check for user being an owenr of any organization and then redirect if not orgless

shakalee14 committed:
Merge branch 'fix-orgcreatejoin' into 1032-admin-login-bug

shakalee14 committed:
Merge branch 'lambda-acceptspecialchartextstests' into stage-main-2

shakalee14 committed:
Merge branch '1032-admin-login-bug' into stage-main-2

lperson committed:
fix https://github.com/MoveOnOrg/Spoke/pull/1025#issuecomment-449762417

shakalee14 committed:
Merge branch 'lperson-wfp_staging_20180902' into stage-main-2

shakalee14 committed:
Merge pull request #1031 from MoveOnOrg/stage-main-2

Stage main 2

harpojaeger committed:
Merge branch 'main' into merge-main

harpojaeger committed:
remove package-lock.json; this should be its own PR

harpojaeger commented:
@schuyler1d, I updated my branch from upstream/main; can you do the same for this PR so I can see the changes in isolation? Thanks!

harpojaeger commented:
(I resolved the merge conflicts, so they should go away once you merge main.)

harpojaeger committed:
Exclude the knex schema equality tests from automatic Jest runs.
These assertions are meant to be run "by hand" to check that the new migration system creates equivalent tables to the old one. As yet, it's not possible to automate them.

harpojaeger committed:
generate correct schema for messages table

harpojaeger committed:
generate the correct campaign table schema

harpojaeger committed:
update timestamp of initial knex migration

bchrobot committed:
End script when complete.

bchrobot committed:
Use writeFileSync.

Use the syncronous version as we are already inside an async promise. This fixes issue @bchrobot experienced where campaign_contact just would not write to disk.

bchrobot committed:
Replace log statement.

harpojaeger committed:
Merge pull request #2 from bchrobot/knex-migrations

export.js fixup

harpojaeger committed:
fix typo

harpojaeger committed:
fix test-knex command so that it runs the correct test even with the jest ignore patterns

harpojaeger committed:
preliminary: use a new raw query to get an actually-accurate list of indices
quote> Tests still do not pass, as they'll need to be adapted to the new format of the returned object, but the number of indices now matches, which is good progress.

harpojaeger committed:
exclude migrations table from key queries

harpojaeger committed:
reorganize the export files

harpojaeger committed:
define index query in one place

harpojaeger committed:
update export script to match new location

harpojaeger committed:
update the index test to use the new pg constraintdef data

harpojaeger committed:
update the index sql query to exclude knex tables

harpojaeger committed:
correctly generate the user_cell foreign key

harpojaeger committed:
add attribution for index query

harpojaeger committed:
define knex in src/server/knex.js and use that instance elsewhere

harpojaeger committed:
Commit initial DB test data to version control.
This makes the tests more replicable, and eliminates the need for 
modifications to jest.config.js.

harpojaeger committed:
Remove old migrations infrastructure

harpojaeger committed:
Remove OSDI fields from initial DB migration.

harpojaeger committed:
Update developer docs to refer to new migration strategy.

harpojaeger committed:
Switch remaining knex import to use the new centrally-defined instance.

harpojaeger committed:
Switch to importing knex from Thinky.
This is necessary because the rethink-knex-adapter can only accept a 
config object, not an existing instance of knex.

harpojaeger committed:
Fix typo.

harpojaeger committed:
Try bumping Travis PG version.

harpojaeger committed:
Correct Travis PG syntax.

harpojaeger committed:
Even more Travis YAML refinements.

harpojaeger committed:
Try with pg 10.0, since Travis doesn't seem to like 11.1.
See https://github.com/travis-ci/travis-ci/issues/5737.

harpojaeger committed:
Go back to default Travis PG version, pending resolution of https://github.com/travis-ci/travis-ci/issues/5737

schuyler1d committed:
merge main and harpo/knex-migrations
